File: 02_algorithm/programmers/Level1/모의고사.py
def solution(answers):
    answer = []
    patterns = [[1, 2, 3, 4, 5], [2, 1, 2, 3, 2, 4, 2, 5], [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]]
    answer_cnt = []
    for pattern in patterns:
        count = 0
        for idx in range(len(answers)):
            if answers[idx] == pattern[idx % len(pattern)]:
                count += 1
        answer_cnt.append(count)
    max_value = max(answer_cnt)
    for person_number in range(3):
        if answer_cnt[person_number] == max_value:
            answer.append(person_number + 1)
    return answer

print(solution([1,2,3,4,5]))
print(solution([1,3,2,4,2]))

File: 06_js/04_algorithm/programmers/Level1/모의고사.js
function solution(answers) {
  let answer = []
  let answer_cnt = []
  const patterns = [
    [1, 2, 3, 4, 5],
    [2, 1, 2, 3, 2, 4, 2, 5],
    [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
  ]
  patterns.forEach(pattern => {
    let count = 0
    let patternCount = pattern.length
    for (let i = 0; i < answers.length; ++i) {
      if (answers[i] === pattern[i % patternCount]) {
        count += 1
      }
    }
    answer_cnt.push(count) 
  })
  const max_value = Math.max.apply(null, answer_cnt)
  for (let i = 0; i < 3; ++i) {
    if (answer_cnt[i] === max_value) {
      answer.push(i + 1)
    }
  }
  return answer
}

console.log(solution([1, 2, 3, 4, 5]))
console.log(solution([1, 3, 2, 4, 2]))